cmd/client.go

```
@@ -0,0 +1,188 @@
+
+// NewClient creates, initialises and returns a new CloudproberClient.
+// This is the correct way to create a new instance of CloudproberClient.
+// Parameters:
+// - rpcServer: Pass the rpcServer address and port as a string formatted as: "hostname:port", e.g. "localhost:1234"
+// Returns:
+// - error:
+//	  - nil: No error occurred and the client was successfully initialised.
+//	  - gRPC Error: See https://godoc.org/google.golang.org/grpc/codes for error codes.
+//		-> See https://github.com/grpc/grpc-go/blob/d25c71b54334380ff1febd25d88064b36de44b3c/clientconn.go#L123
+func NewClient(rpcServer string) (*CloudproberClient, error) {
+	conn, err := grpc.Dial(rpcServer, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("GRPC_DIAL_ERROR: %v", err)
+	}
+	return &CloudproberClient{conn: conn, client: proberpb.NewCloudproberClient(conn)}, nil
+}
+
+// initClient establishes an active client connection to the Cloudprober gRPC server.
+// If there is already an active connection, it will do nothing and return nil.
+// If there is not an active connection, it will make one and assign a Cloudprober client
+// to the CloudproberClient.client field.
+// Parameters:
+// - rpcServer: Pass the rpcServer address and port as a string formatted as: "hostname:port", e.g. "localhost:1234"
+// Returns:
+// - error:
+//	  - nil: No error occurred and the client was successfully initialised.
+//	  - gRPC Error: See https://godoc.org/google.golang.org/grpc/codes for error codes.
+//		-> See https://github.com/grpc/grpc-go/blob/d25c71b54334380ff1febd25d88064b36de44b3c/clientconn.go#L123
+func (client *CloudproberClient) initClient(rpcServer string) error {
+	client.clientMux.Lock()
+	defer client.clientMux.Unlock()
+	var err error
+
+	if client.client == nil { // If there is not an active client connection, make one.
+		client.conn, err = grpc.Dial(rpcServer, grpc.WithInsecure())
+		if err != nil {
+			dialErr := fmt.Errorf("GRPC_DIAL_ERROR: %v", err)
+			return dialErr
```

why not just return the error directly w/o needing to define a new variable?
```suggestion
			return fmt.Errorf("GRPC_DIAL_ERROR: %v", err)
```

---

This is addressed in a more recent commit.